crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheReader.java

```diff
@@ -0,0 +1,56 @@
+package com.google.cloud.tools.crepecake.cache;
+
+import com.google.cloud.tools.crepecake.image.ImageLayers;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.util.Set;
+
+/** Reads image content from the cache. */
+public class CacheReader {
+
+  private final Cache cache;
+
+  public CacheReader(Cache cache) {
+    this.cache = cache;
+  }
+
+  /**
+   * Gets the file that stores the content BLOB for an application layer.
+   *
+   * @param layerType the type of layer
+   * @param sourceFiles the source files the layer must be built from
+   * @return
+   * @throws CacheMetadataCorruptedException
+   */
+  public File getLayerFile(CachedLayerType layerType, Set<Path> sourceFiles)
+      throws CacheMetadataCorruptedException {
+    switch (layerType) {
+      case DEPENDENCIES:
+      case RESOURCES:
+      case CLASSES:
+        CacheMetadata cacheMetadata = cache.getMetadata();
```

Comment:
Does the caller of getLayerFile know that their layerType is not supported? Or is the expectation that any layer can be supported.

I guess this comes down to a question of whether it's completely necessary to differentiate layers based on TYPE since they might all function in the same way?

Reply:
Yeah, this is intended to be only used for application layers. Base image layers will not have a sourceFiles property (since they aren't built from source files) and thus do not fit into the purpose of this method. I changed the wording of the comment a bit to explain this better.

Reply:
Actually, I'll just add an UnsupportedOperationException.

Reply:
I think the question is really that is type truly necessary to pass into this method?

Reply:
It is necessary to filter the cached layers in the metadata (line 35).